services/power.md

```diff
@@ -4,6 +4,69 @@
 
 A power-provider service.
 
+## Power negotiation protocol
+
+The purpose of the power negotiation is to ensure that there is no more than ~500mA
+delivered to the power rail.
+This is realized by limiting the number of enabled power provider services to 1.
+
+Note, that it's also possible to have low-current power providers, which do not
+run a power provider service.
+These are **not** accounted for in the power negotiation protocol.
+
+The protocol is based on `on` reports, which are always sent 
+after general device announce packets, in the same frame.
+This makes it simpler for other power services to parse them.
+
+The `on` reports contain device priority, which is either its remaining battery
+capacity, or a constant.
+
+After queuing an announce with `on` report, the service enters a grace period
+until the report has been sent on the wire.
+During grace period incoming `on` reports are ignored.
+
+* Upon reset, a power service enables itself (instant-on), and after 0-300ms (random)
+  send the first announce packet (with `on` report)
+* Every enabled power service emits power `on` reports with its announce packets,
+  which are sent every 400-600ms (random; first few announce packets can be even sent more often)
+* If an enabled power service sees a power `on` report from somebody else of higher or equal priority,
```

Comment:
priority has not been introducted yet here.

Reply:
it's defined a few lines above